src/components/UserFormDetails/index.js

```diff
@@ -312,16 +314,87 @@ export const UserFormDetails = (props) => {
     }
   }
 
+  const updatePromotions = async (change, setState, state) => {
+    try {
+      let response
+        response = await ordering.users(props?.userData?.id || userState.result.result.id).save(change, {
+          accessToken: accessToken
+        })
+        setState({
+          ...state,
+          changes: response.content.error ? change : {},
+          result: response.content,
+          loading: false
+        })
+
+      if (!response.content.error) {
+        setUserState({
+          ...userState,
+          result: {
+            ...userState.result,
+            ...response.content
+          }
+        })
+        if (!isCustomerMode) {
+          changeUser({
+            ...session.user,
+            ...response.content.result
+          })
+        } else {
+          setUserCustomer({
+            ...customer.user,
+            ...response.content.result
+          }, changes?.setCustomerInLocal ?? true)
+        }
+
+        if (handleSuccessUpdate) {
+          handleSuccessUpdate(response.content.result)
+        }
+      }
+    } catch (err) {
+      setState({
+        ...state,
+        result: {
+          error: true,
+          result: err.message
+        },
+        loading: false
+      })
+    }
+  }
+
   const handleChangePromotions = (enabled) => {
-    setFormState({
-      ...formState,
+    setNotificationsGroup({
+      ...notificationsGroup,
       changes: {
-        ...formState.changes,
-        settings: { notification: { newsletter: enabled, promotions: enabled }, sms: { newsletter: enabled, promotions: enabled } }
-      }
+        ...notificationsGroup?.changes,
+        settings: { email: { newsletter: enabled, promotions: enabled }, notification: { newsletter: enabled, promotions: enabled }, sms: { newsletter: enabled, promotions: enabled } }
+      },
+      loading: true
     })
   }
+  
+  const handleChangeNotifications = (value) => {
```

Comment:
You can make only one method and using a prop you can switch between the 2 states